pkg/syncer/syncertest/fake/storage.go

```diff
@@ -34,17 +34,34 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
 	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/strategicpatch"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/klog/v2"
 	"kpt.dev/configsync/pkg/api/configsync"
 	"kpt.dev/configsync/pkg/core"
 	"kpt.dev/configsync/pkg/kinds"
+	"kpt.dev/configsync/pkg/reconcilermanager"
 	"kpt.dev/configsync/pkg/util/log"
 	"sigs.k8s.io/cli-utils/pkg/testutil"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	// FieldManager is the field manager used by NewClient, when initializing
+	// objects on the cluster.
+	// This avoids conflicts between the reconciler and reconciler-manager.
+	FieldManager = configsync.ConfigSyncPrefix + "fake-client"
+)
+
+var (
+	defaultAllowedFieldManagers = sets.New[string](
+		FieldManager,
+		configsync.FieldManager,
+		reconcilermanager.FieldManager,
```

Comment:
Should the nomos-cli and resource-group-controller field manager be included?

Reply:
They could be added in the future, but there aren't any unit tests for those components that use the fake clients here.